sklearn/gaussian_process/gpc.py

```diff
@@ -0,0 +1,197 @@
+
+# Authors: Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+#
+# License: BSD 3 clause
+
+import numpy as np
+from scipy.linalg import cholesky, cho_solve, solve
+from scipy.optimize import fmin_l_bfgs_b
+from scipy.special import erf
+
+from sklearn.base import BaseEstimator, clone
+from sklearn.gaussian_process.kernels import RBF
+from sklearn.utils.validation import check_X_y, check_is_fitted, check_array
+
+
+# Values required for approximating the logistic sigmoid by
+# error functions. coefs are obtained via:
+# x = np.array([0, 0.6, 2, 3.5, 4.5, np.inf])
+# b = logistic(x)
+# A = (erf(np.dot(x, self.lambdas)) + 1) / 2
+# coefs = lstsq(A, b)[0]
+LAMBDAS = np.array([0.41, 0.4, 0.37, 0.44, 0.39])[:, np.newaxis]
+COEFS = np.array([-1854.8214151, 3516.89893646, 221.29346712,
+                  128.12323805, -2010.49422654])[:, np.newaxis]
+
+
+class GaussianProcessClassifier(BaseEstimator):
+    """ Gaussian process classification (GPC).
+
+    The implementation is based on Algorithm 3.1, 3.2, and 5.1 of
+    ``Gaussian Processes for Machine Learning'' (GPML) by Rasmussen and
+    Williams.
+
+    Internally, the Laplace approximation is used for approximating the
+    non-Gaussian posterior by a Gaussian.
+
+    Currently, the implementation is restricted to
+      * using the logistic link function
+      * binary classification
+    """
+
+    def __init__(self, kernel=None, jitter=0.0, optimizer="fmin_l_bfgs_b"):
+        self.kernel = kernel
+        self.jitter = jitter
+        self.optimizer = optimizer
+
+    def fit(self, X, y):
+        if self.kernel is None:  # Use an RBF kernel as default
+            self.kernel_ = RBF()
+        else:
+            self.kernel_ = clone(self.kernel)
+
+        X, y = check_X_y(X, y)
+
+        # XXX: Assert that y is binary and labels are {0, 1}
+        self.X_fit_ = X
+        self.y_fit_ = check_array(y, ensure_2d=False, dtype=float)
+
+        if self.optimizer == "fmin_l_bfgs_b":
+            # Choose hyperparameters based on maximizing the log-marginal
+            # likelihood
+            def obj_func(theta):
+                lml, grad = self.log_marginal_likelihood(theta,
+                                                         eval_gradient=True)
+                return -lml, -grad
+            self.theta_, _, _ = fmin_l_bfgs_b(obj_func, self.kernel_.theta,
+                                              bounds=self.kernel_.bounds)
+            self.kernel_.theta = self.theta_
+        elif self.optimizer is None:
+            self.theta_ = self.kernel_.theta
+        else:
+            raise ValueError("Unknown optimizer %s." % self.optimizer)
+
+        # Precompute quantities required for predictions which are independent
+        # of actual query points
+        self.K_ = self.kernel_(self.X_fit_)
+        self.K_[np.diag_indices_from(self.K_)] += self.jitter
+
+        self.f_, _, (self.pi, self.W_sr, self.L, _, _) = \
```

Comment:
persisted values that have been calculated should end in `_`, i.e. `self.pi_`, `self.W_sr_`